project/CMakeLists.txt

```diff
@@ -31,31 +31,33 @@ if (NOT link_libraries)
 endif()
 
 if (NOT ${project_config_recursive_file_gathering})
-    configure_subprojects(${PROJECT_NAME} link_targets)
+    configure_subprojects(${PROJECT_NAME} link_targets ${project_config_unit_tests_file_tag})
 endif()
 
 ### Gather files and put them in folders
-set(file_excludes ".*\\.tests.*" ".*\\.bench.*")
+set(file_excludes ".*\\${project_config_unit_tests_file_tag}.*" ".*\\${project_config_benchmark_file_tag}.*")
 fill_platform_excludes(platform_excludes)
 list(APPEND file_excludes ${platform_excludes})
 gather_files(header_files ${project_config_recursive_file_gathering} "../include/${project_config_name}/*.hpp;../include/${project_config_name}/*.h;../include/${project_config_name}/*.inl;../src/*.hpp;../src/*.h;../src/*.inl" "${file_excludes}")
-gather_files(src_files ${project_config_recursive_file_gathering} "../src/*.cpp" "${file_excludes}")
+gather_files(src_files ${project_config_recursive_file_gathering} "../src/${project_config_name}/*.cpp" "${file_excludes}")
+list(APPEND src_files "../src/main.cpp")
 
 if (project_config_extra_sources)
     list(APPEND src_files "${project_config_extra_sources}")
 endif()
 
 if (${project_config_use_unit_tests})
-    set(file_patterns "../src/*.tests.cpp")
+    set(file_patterns "../tests/*.tests.cpp")

```

Comment:
#should be using `${project_config_unit_tests_file_tag}` instead of the hardcoded `.tests` string